sound/soc/sof/intel/hda-loader.c

```diff
@@ -470,3 +471,26 @@ int hda_dsp_post_fw_run(struct snd_sof_dev *sdev)
 	/* re-enable clock gating and power gating */
 	return hda_dsp_ctrl_clock_power_gating(sdev, true);
 }
+
+int hda_dsp_ext_man_get_cavs_config_data(struct snd_sof_dev *sdev,
+					 const struct sof_ext_man_elem_header *hdr)
+{
+	const struct sof_ext_man_cavs_config_data *config_data =
+		container_of(hdr, struct sof_ext_man_cavs_config_data, hdr);
+	struct sof_intel_hda_dev *hda = sdev->pdata->hw_pdata;
+	int i;
+
+	for (i = 0; i < SOF_EXT_MAN_CAVS_CONFIG_LAST_ELEM; i++)
+		switch (config_data->elems[i].token) {
+		case SOF_EXT_MAN_CAVS_CONFIG_CAVS_LPRO:
+			hda->clk_config = config_data->elems[i].value;
+			dev_dbg(sdev->dev, "FW clock config: %s\n", hda->clk_config ? "LPRO" : "HPRO");
+			break;
+		default:
+			dev_warn(sdev->dev, "unsupported token type: %d\n",
```

Comment:
orshould this be an error?

Reply:
Yes. you're right. This would be error now.(it wasn't). Will update this

Reply:
@ranj063 @fredoh9 Can you expand why you think this should be an error? I'd actually rather have this as dev_dbg(). We are committed to maintain the ABI, so it's basicly expected case that a newer, but backwards compatible FW is run with an older kernel. With above warning, user will start seeing warnings in kernel, which seems unnecessary. Or am I missing something?